build.py

```diff
@@ -56,7 +56,8 @@
     "i386": None,
     "x86": None,
     "x86_64": None,
-    "arc": "arc-linux-"
+    "arc": "arc-linux-",
+    "riscv": "riscv64-linux-gnu-",
```

Comment:
This needs to be `riscv64` to match the Docker image names and the ARCH name used in the builds.

Reply:
Second thought: the kernel only knows about `riscv` as an architecture.  So we have a problem :)

Option 1: we use `riscv` in the Docker images instead of `riscv64`
Option 2: we sort out build configuration properly if we really need different RISC-V toolchains

Do we actually need to build for multiple RISC-V variants?  Also, is the kernel arch `riscv` actually `riscv64`?

Reply:
Except we're not going to patch the kernel to move arch/riscv as arch/riscv64 :p  or are we?

So we probably need another kind of hack to call `build.py` with `ARCH=riscv` while everything else will be using `riscv64`.

Reply:
> Except we're not going to patch the kernel to move arch/riscv as arch/riscv64 :p or are we?

I was editing my comment to add the last words while you were leaving this comment :P

>  So we probably need another kind of hack to call build.py with ARCH=riscv while everything else will be using riscv64.

this seems the best solution to me.

Reply:
I think this is the same problem with mips, the kernel calls it "mips" but you need >1 compilers to build all the defconfigs, which debian calls mips, mipsel, mips64el and possibly more.
Wasnt there a solution to this suggest when it was agreed to split x86 into x86_64 and i386?

Reply:
For x86 it's not the same issue as the same host toolchain can be used to build either i386 or x86_64 code.

The issue with x86 was that you can't distinguish the user-space variants, so we really need to use i386 or x86_64 instead.

Reply:
IMO, build.py is for the kernel, so it needs to use the arch value known by the kernel, and map to a (default) toolchain, so I think this is the best we can do.

Note that this is only the default for build.py, if $CROSS_COMPILE is set in the caller env, that will override this build.py value anyways.